prover/src/lib.rs

```diff
@@ -384,28 +395,24 @@ pub trait Prover {
         // compute random linear combinations of these evaluations using coefficients drawn from
         // the channel
         let ce_domain_size = air.ce_domain_size();
-        let composition_poly_trace =
-            info_span!("evaluate_constraints", ce_domain_size).in_scope(|| {
-                self.new_evaluator(
-                    &air,
-                    aux_rand_elements,
-                    channel.get_constraint_composition_coeffs(),
-                )
-                .evaluate(&trace_lde, &domain)
-            });
+        let composition_poly_trace = self
+            .new_evaluator(&air, aux_rand_elements, channel.get_constraint_composition_coeffs())
+            .await
+            .evaluate(&trace_lde, &domain);
```

Comment:
Fixed by instrumenting `DefaultConstraintEvaluator::evaluate()`, and configuring it so that it preserves the same output as currently. The downside is that every new evaluator will need to be instrumented separately, but I don't know a way around it

Reply:
Very minor: we controlled the name of the span with `in_scope()`, but now we rely on the implementor to format the span in a similar way (if even needed/desired?)